eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -109,6 +111,13 @@ jobs:
               artifactName: 'artifacts' 
               targetPath: $(Build.ArtifactStagingDirectory)
 
+          - task: PythonScript@0
```

Comment:
We only need to set the dev version on the legs where we produce the packages. For the testing is isn't necessary.

Reply:
Testing step ensures we are testing correct wheel by checking for a whl with same version is present in artifacts. I have added the change to update version here so that build test yml and tox methods for tests will not not be affected. Alternate option i had thought is it skip verifying whl in artifacts in case of dev build but this will need dev build flag passed down to tox harness code.

Current flow as per my understanding is:
1. build packages
2. Test phase creates whl and sdist and making sure there is a whl and sdist with same name in downloaded artifacts. ( This check fails if i don't update the version before test).
@scbedd: Please let me know if my understanding is wrong.

I have tested both scenarios with and without version changes in test phase and test fails if i update only when building package.

Reply:
If we need to set the dev version in the test legs i'm fine with that, but we should do it in a consistent way. I chatted with @mitchdenny and he is going to create a template that we can use to setup the SetDevVersion property that will ensure we all do it in a consistent manner.